public/app/features/dashboard-scene/serialization/annotations.ts

```diff
@@ -0,0 +1,124 @@
+
+  dsType: string,
+  dsUID?: string,
+  // Overrides are used to provide properties based on scene's annotations data layer object state
+  override?: Partial<AnnotationQuery>
+): AnnotationQueryKind {
+  const group = annotation.builtIn ? 'grafana' : dsType;
+
+  const {
+    // known properties documented in v1 schema
+    enable,
+    hide,
+    iconColor,
+    name,
+    builtIn,
+    filter,
+    mappings,
+    datasource,
+    target,
+    snapshotData,
+    type,
+
+    // unknown properties that are still available for configuration through API
+    ...legacyOptions
+  } = annotation;
+
+  const result: AnnotationQueryKind = {
+    kind: 'AnnotationQuery',
+    spec: {
+      builtIn: Boolean(annotation.builtIn),
+      name: annotation.name,
+      enable: Boolean(override?.enable) || Boolean(annotation.enable),
+      hide: Boolean(override?.hide) || Boolean(annotation.hide),
+      iconColor: annotation.iconColor,
+
+      query: {
+        kind: 'DataQuery',
+        version: defaultDataQueryKind().version,
+        group, // Annotation layer has a datasource type provided in runtime.
+        spec: target || {},
+      },
+    },
+  };
+
+  if (dsUID) {
+    result.spec.query.datasource = {
+      name: dsUID,
+    };
+  }
+
+  // if legacy options is not an empty object, add it to the result
+  if (Object.keys(legacyOptions).length > 0) {
+    result.spec.legacyOptions = legacyOptions;
+  }
+
+  if (annotation.filter?.ids?.length) {
+    result.spec.filter = annotation.filter;
+  }
+
+  // TODO: add mappings
+
+  return result;
+}
+
+export function transformV2ToV1AnnotationQuery(annotation: AnnotationQueryKind): AnnotationQuery {
+  let { query: dataQuery, ...annotationQuery } = annotation.spec;
+
+  // Mapping from AnnotationQueryKind to AnnotationQuery used by scenes.
+  let annoQuerySpec: AnnotationQuery = {
+    enable: annotation.spec.enable,
+    hide: annotation.spec.hide,
+    iconColor: annotation.spec.iconColor,
+    name: annotation.spec.name,
+    // TOOO: mappings
+  };
+
+  if (Object.keys(dataQuery.spec).length > 0) {
+    // @ts-expect-error DataQueryKind spec should be typed as DataQuery interface
+    annoQuerySpec.target = {
+      ...dataQuery?.spec,
+    };
+  }
+
+  if (annotation.spec.builtIn) {
+    annoQuerySpec.type = 'dashboard';
+    annoQuerySpec.builtIn = 1;
+  }
+
+  if (annotation.spec.filter) {
+    annoQuerySpec.filter = annotation.spec.filter;
+  }
+
+  //   THIS DOES NOT MAKE SENSE< AnnotationQuery would never have this property
+  // some annotations will contain in the legacyOptions properties that need to be
+  // added to the root level annotation spec
+  if (annotationQuery.legacyOptions) {
+    annoQuerySpec = {
+      ...annoQuerySpec,
+      ...annotationQuery.legacyOptions,
+
+      //   THIS DOES NOT MAKE SENSE< AnnotationQuery would never have this property
+      //   legacyOptions: {
+      //     ...annotationQuery.legacyOptions,
+      //   },
+    };
+  }
```

Comment:
Reverting a stupid change I've made in https://github.com/grafana/grafana/pull/104194/commits/c74e78794be96aa4d7ee34460983f712e6245fd5